mmda/predictors/heuristic_predictors/dictionary_word_predictor.py

```diff
@@ -0,0 +1,180 @@
+import string
+from typing import Optional
+
+from mmda.predictors.base_predictors.base_predictor import BasePredictor
+from mmda.types.annotation import Annotation, Span, SpanGroup
+from mmda.types.document import Document
+from mmda.types.names import Rows, Tokens
+
+
+class DictionaryWordPredictor(BasePredictor):
+
+    REQUIRED_BACKENDS = None
+    REQUIRED_DOCUMENT_FIELDS = [Rows, Tokens]
+
+    _dictionary: Optional[set[str]] = None
+
+    def __init__(self, dictionary_file_path: str) -> None:
+        """Build a predictor that indexes the given dictionary file.
+        A dictionary is simply a case-sensitive list of words as a text file.
+        Words should be lower-case in the dictionary unless they are invalid
+        as a lower-case word (e.g., a person's name).
+
+        For an example, see https://github.com/dwyl/english-words (words_alpha.txt)
```

Comment:
I think we could cache it to s2 aws and if the dictionary_file_path is not set, we can automatically have the default version downloaded.

Reply:
I think that giving a reference here is the right thing. This way the consume can decide if the license of the consumed dictionary is appropriate. Currently the example repository is Unlicense which would be fine in general, but there is an underlying reference to the original copyright holder being someone else. So, I think users should be able to decide what dictionary reference to use.